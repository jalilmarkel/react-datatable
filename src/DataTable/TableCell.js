import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import styled, { withTheme } from 'styled-components';
import { getProperty } from './util';

const TableCellStyle = styled.td`
  box-sizing: border-box;
  vertical-align: middle;
  line-height: normal;
  white-space: nowrap;
  font-size: ${props => props.theme.rows.fontSize};
  font-weight: 400;
  color: ${props => props.theme.rows.fontColor};
  height: ${props => props.theme.rows.height};
  width: ${props => props.column.width};
  ${props => props.column.number && 'text-align: right'};
  ${props => props.column.center && 'text-align: center'};
  padding-left: calc(${props => props.theme.cells.cellPadding} / 2);
  padding-right: calc(${props => props.theme.cells.cellPadding} / 2);

  &:nth-child(n+2) {
    padding-left: calc(${props => props.theme.cells.cellPadding} / 6);
  }

  ${props => props.column.compact && 'padding: 0'};
`;

class TableCell extends PureComponent {
  static propTypes = {
    column: PropTypes.object,
    row: PropTypes.object,
  };

  static defaultProps = {
    column: {},
    row: {},
  };

  handleCellClick = e => {
    const { column } = this.props;

    if (column.ignoreRowClick) {
      e.stopPropagation();
    }
  };

  render() {
    const {
      column,
      row,
    } = this.props;

    return (
      <TableCellStyle
        column={column}
        onClick={this.handleCellClick}
      >
        {column.cell ? column.cell(row) : getProperty(row, column.selector, column.format)}
      </TableCellStyle>
    );
  }
}

export default withTheme(TableCell);
