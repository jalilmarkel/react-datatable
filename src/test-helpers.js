import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import 'jest-styled-components';
import theme from './themes/default';

export const renderWithTheme = (tree, ...args) =>
  render(<ThemeProvider theme={theme()}>{tree}</ThemeProvider>, ...args);

