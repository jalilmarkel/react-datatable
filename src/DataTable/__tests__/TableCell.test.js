import 'jest-styled-components';

import React from 'react';
import { shallowWithTheme } from '../../test-helpers';

import TableCell from '../TableCell';

test('component <TableCell /> should render correctly', () => {
  const wrapper = shallowWithTheme(<TableCell row={{}} column={{ selector: 'test' }} />);

  expect(wrapper.dive().dive()).toMatchSnapshot();
});
