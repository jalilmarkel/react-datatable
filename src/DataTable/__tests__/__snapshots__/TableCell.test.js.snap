// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableCell /> should render correctly if column.allowOverflow 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  line-height: normal;
  padding-left: calc(48px / 6);
  padding-right: calc(48px / 6);
  padding-top: calc(48px / 12);
  padding-bottom: calc(48px / 12);
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  max-width: 100%;
  min-width: 100px;
  font-size: 13px;
  font-weight: 400;
  white-space: nowrap;
  min-height: 48px;
}

.c0:first-child {
  padding-left: calc(48px / 2);
}

.c0:last-child {
  padding-right: calc(48px / 2);
}

.c0 .react-data-table--cell-content {
  color: inherit;
  overflow: visible;
  text-overflow: ellipsis;
}

<div
  class="rdt_TableCell c0"
>
  <div
    class="react-data-table--cell-content"
  >
    testname
  </div>
</div>
`;

exports[`<TableCell /> should render correctly if column.button 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  line-height: normal;
  padding-left: calc(48px / 6);
  padding-right: calc(48px / 6);
  padding-top: calc(48px / 12);
  padding-bottom: calc(48px / 12);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  max-width: 100%;
  min-width: 100px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 13px;
  font-weight: 400;
  white-space: nowrap;
  min-height: 48px;
}

.c0:first-child {
  padding-left: calc(48px / 2);
}

.c0:not(:last-child) {
  padding-right: calc(48px / 2);
}

.c0 .react-data-table--cell-content {
  color: inherit;
  overflow: hidden;
  text-overflow: ellipsis;
}

<div
  class="rdt_TableCell c0"
>
  <div
    class="react-data-table--cell-content"
  >
    testname
  </div>
</div>
`;

exports[`<TableCell /> should render correctly if column.center 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  line-height: normal;
  padding-left: calc(48px / 6);
  padding-right: calc(48px / 6);
  padding-top: calc(48px / 12);
  padding-bottom: calc(48px / 12);
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  max-width: 100%;
  min-width: 100px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 13px;
  font-weight: 400;
  white-space: nowrap;
  min-height: 48px;
}

.c0:first-child {
  padding-left: calc(48px / 2);
}

.c0:last-child {
  padding-right: calc(48px / 2);
}

.c0 .react-data-table--cell-content {
  color: inherit;
  overflow: hidden;
  text-overflow: ellipsis;
}

<div
  class="rdt_TableCell c0"
>
  <div
    class="react-data-table--cell-content"
  >
    testname
  </div>
</div>
`;

exports[`<TableCell /> should render correctly if column.maxWidth 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  line-height: normal;
  padding-left: calc(48px / 6);
  padding-right: calc(48px / 6);
  padding-top: calc(48px / 12);
  padding-bottom: calc(48px / 12);
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  max-width: 200px;
  min-width: 100px;
  font-size: 13px;
  font-weight: 400;
  white-space: nowrap;
  min-height: 48px;
}

.c0:first-child {
  padding-left: calc(48px / 2);
}

.c0:last-child {
  padding-right: calc(48px / 2);
}

.c0 .react-data-table--cell-content {
  color: inherit;
  overflow: hidden;
  text-overflow: ellipsis;
}

<div
  class="rdt_TableCell c0"
>
  <div
    class="react-data-table--cell-content"
  >
    testname
  </div>
</div>
`;

exports[`<TableCell /> should render correctly if column.minWidth 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  line-height: normal;
  padding-left: calc(48px / 6);
  padding-right: calc(48px / 6);
  padding-top: calc(48px / 12);
  padding-bottom: calc(48px / 12);
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  max-width: 100%;
  min-width: 200px;
  font-size: 13px;
  font-weight: 400;
  white-space: nowrap;
  min-height: 48px;
}

.c0:first-child {
  padding-left: calc(48px / 2);
}

.c0:last-child {
  padding-right: calc(48px / 2);
}

.c0 .react-data-table--cell-content {
  color: inherit;
  overflow: hidden;
  text-overflow: ellipsis;
}

<div
  class="rdt_TableCell c0"
>
  <div
    class="react-data-table--cell-content"
  >
    testname
  </div>
</div>
`;

exports[`<TableCell /> should render correctly if column.right 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  line-height: normal;
  padding-left: calc(48px / 6);
  padding-right: calc(48px / 6);
  padding-top: calc(48px / 12);
  padding-bottom: calc(48px / 12);
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  max-width: 100%;
  min-width: 100px;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  font-size: 13px;
  font-weight: 400;
  white-space: nowrap;
  min-height: 48px;
}

.c0:first-child {
  padding-left: calc(48px / 2);
}

.c0:last-child {
  padding-right: calc(48px / 2);
}

.c0 .react-data-table--cell-content {
  color: inherit;
  overflow: hidden;
  text-overflow: ellipsis;
}

<div
  class="rdt_TableCell c0"
>
  <div
    class="react-data-table--cell-content"
  >
    testname
  </div>
</div>
`;

exports[`<TableCell /> should render correctly if column.width 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  line-height: normal;
  padding-left: calc(48px / 6);
  padding-right: calc(48px / 6);
  padding-top: calc(48px / 12);
  padding-bottom: calc(48px / 12);
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  max-width: 100%;
  min-width: 100px;
  min-width: 200px;
  max-width: 200px;
  font-size: 13px;
  font-weight: 400;
  white-space: nowrap;
  min-height: 48px;
}

.c0:first-child {
  padding-left: calc(48px / 2);
}

.c0:last-child {
  padding-right: calc(48px / 2);
}

.c0 .react-data-table--cell-content {
  color: inherit;
  overflow: hidden;
  text-overflow: ellipsis;
}

<div
  class="rdt_TableCell c0"
>
  <div
    class="react-data-table--cell-content"
  >
    testname
  </div>
</div>
`;

exports[`<TableCell /> should render correctly if column.wrap 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  line-height: normal;
  padding-left: calc(48px / 6);
  padding-right: calc(48px / 6);
  padding-top: calc(48px / 12);
  padding-bottom: calc(48px / 12);
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  max-width: 100%;
  min-width: 100px;
  font-size: 13px;
  font-weight: 400;
  white-space: normal;
  min-height: 48px;
}

.c0:first-child {
  padding-left: calc(48px / 2);
}

.c0:last-child {
  padding-right: calc(48px / 2);
}

.c0 .react-data-table--cell-content {
  color: inherit;
  overflow: hidden;
  text-overflow: ellipsis;
}

<div
  class="rdt_TableCell c0"
>
  <div
    class="react-data-table--cell-content"
  >
    testname
  </div>
</div>
`;

exports[`<TableCell /> should render correctly if not an internal cell (e.g. checkbox or expander 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  line-height: normal;
  padding-left: calc(48px / 6);
  padding-right: calc(48px / 6);
  padding-top: calc(48px / 12);
  padding-bottom: calc(48px / 12);
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  max-width: 100%;
  min-width: 100px;
  font-size: 13px;
  font-weight: 400;
  white-space: nowrap;
  min-height: 48px;
}

.c0:last-child {
  padding-right: calc(48px / 2);
}

.c0 .react-data-table--cell-content {
  color: inherit;
  overflow: hidden;
  text-overflow: ellipsis;
}

<div
  class="rdt_TableCell c0"
>
  <div
    class="react-data-table--cell-content"
  >
    testname
  </div>
</div>
`;

exports[`<TableCell /> should render correctly with defaults 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  line-height: normal;
  padding-left: calc(48px / 6);
  padding-right: calc(48px / 6);
  padding-top: calc(48px / 12);
  padding-bottom: calc(48px / 12);
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  max-width: 100%;
  min-width: 100px;
  font-size: 13px;
  font-weight: 400;
  white-space: nowrap;
  min-height: 48px;
}

.c0:first-child {
  padding-left: calc(48px / 2);
}

.c0:last-child {
  padding-right: calc(48px / 2);
}

.c0 .react-data-table--cell-content {
  color: inherit;
  overflow: hidden;
  text-overflow: ellipsis;
}

<div
  class="rdt_TableCell c0"
>
  <div
    class="react-data-table--cell-content"
  >
    testname
  </div>
</div>
`;
